pkg/sql/logictest/logic_test.go

```diff
@@ -314,6 +423,30 @@ func (l *lineScanner) Scan() bool {
 	return ok
 }
 
+func (l *lineScanner) Text() string {
+	text := l.Scanner.Text()
+	if len(l.varMap) == 0 {
+		return text
+	}
+
+	// See if there are any $vars to replace.
```

Comment:
You could do this whole thing with a regex couldn't you? Might be a little more concise, but it's up to you.